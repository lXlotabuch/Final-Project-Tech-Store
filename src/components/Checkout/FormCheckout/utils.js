/* eslint-disable import/prefer-default-export */
export const layoutFormCheckout = {
  labelCol: {
    xs: {
      span: 24,
    },
    sm: {
      span: 6,
    },
    md: {
      span: 8,
    },
    lg: {
      span: 7,
    },
  },
  wrapperCol: {
    xs: {
      span: 24,
    },
    sm: {
      span: 16,
    },
    md: {
      span: 16,
    },
    lg: {
      span: 12,
    },
  },
}