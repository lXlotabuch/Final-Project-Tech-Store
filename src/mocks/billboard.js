/* eslint-disable import/prefer-default-export */
export const billboardMock = {
  title: 'gaming monitors',
  img: 'https://fuzzmusic.ru/wp-content/uploads/2020/08/pesni-iz-igry-kiberpank-2077-1.jpg',
  url: '/catalog?categories=gamingMonitors',
  category: 'gaming monitors'
}