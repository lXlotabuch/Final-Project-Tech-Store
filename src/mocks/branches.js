/* eslint-disable import/prefer-default-export */
export const branchesMock = [
  {
    branchName: 'Branch name',
    branchRef: 'kjhkjh3j23kjhkjhk32jhkj4',
  }
]