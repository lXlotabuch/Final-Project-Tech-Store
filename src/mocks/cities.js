/* eslint-disable import/prefer-default-export */
export const citiesMock = [
  {
    CityName: 'Kyiv',
    Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Dnipro',
    Ref: 'db5c88f0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Kharkiv',
    Ref: 'db5c88e0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Lviv',
    Ref: 'db5c88f5-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Odessa',
    Ref: 'db5c88d0-391c-11dd-90d9-001a92567626'
  },
]

export const cityMock = {
  CityName: 'Kyiv',
  Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
}