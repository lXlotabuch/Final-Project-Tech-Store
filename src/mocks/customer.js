export const mockCustomer = {
  customerNo: '82724063',
  date: '2021-02-19T17:12:50.176+00:00',
  _id: '602ff192ad29b70015add9ab',
  email: 'o.titova.ua@gmail.com',
  isAdmin: false,
  enabled: true,
  firstName: 'someName',
  lastName: 'someLastName',
  login: 'admin',
  password: '$2a$10$i7f6tUO3IT.YsrEGzo/6neTIzoxjhhpWh3Hw3Cme3KLmIAeeyKtFS',
  telephone: '+380731005009',
  gender: 'male',
  ratedProducts: [
    '33823',
    '296354',
    '996598',
    '500227',
    '694186',
    '73109',
    '394414',
    '833228',
    '562546',
    '954601',
    '179158',
    '132036'
  ],
}
export default mockCustomer
