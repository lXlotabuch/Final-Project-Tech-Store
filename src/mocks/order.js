/* eslint-disable import/prefer-default-export */
export const orderMock = {
  deliveryAddress: {country: 'Ukraine', city: 'db5c88f0-391c-11dd-90d9-001a92567626', branch: 'Отделение №2: ул. Академика Янгеля, 40'},
  email: 'o.titova.ua@gmail.com',
  firstName: 'test',
  letterHtml: '<h1>Your order is placed. Our manager will contact you soon.</h1>',
  letterSubject: 'test, thank you for order!',
  mobile: '+380660788828',
  orderNo: '8395086',
  paymentInfo: 'Payment at the time of receipt of the goods',
  shipping: 121,
  totalSum: 117400,
  date: '2021-04-13T14:14:27.478Z',
  _id: '6070d130058f4e00154bbe23'
}