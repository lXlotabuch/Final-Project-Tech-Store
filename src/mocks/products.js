export const productMock = {
  _id: '604ffa554e7ef500153393f9',
  name: 'Monitor 27" AOC C27G2ZU/BK',
  enabled: true,
  currentPrice: 10600,
  previousPrice: 12600,
  categories: 'gamingMonitors',
  imageUrls: [
    'https://i8.rozetka.ua/goods/18842039/aoc_c27g2zu_bk_images_18842039923.jpg',
    'https://i2.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040211.jpg',
    'https://i8.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040937.jpg',
    'https://i8.rozetka.ua/goods/18842041/aoc_c27g2zu_bk_images_18842041729.jpg'
  ],
  quantity: 44,
  color: 'black',
  productUrl: '/gamingMonitors',
  brand: 'AOC',
  newProduct: 'no',
  reviews: [4, 2, 3, 5, 4, 5, 3, 4, 3, 5, 5, 4, 5],
  params: {
    updateFrequency: '240Hz',
    countryOfmanufacture: 'China',
    displayDiagonal: '27"',
    displayResolution: '1920x1080',
    responseTime: '0,5ms',
    screenSize: '32 inches',
    aspectRatio: '16:9'
  },
  description: 'An immersive 27" super curved gaming monitor with a fast 144Hz refresh rate and 1ms response time, helping you get the most out of your monitor. Features such as AMD FreeSync Premium, QHD resolution, and a curved design make this monitor ultra competitive for any gamer.',
  itemNo: '554194',
}
const productMock2 = {
  ...productMock,
  itemNo: '554195',
  _id: '604ffa554e7ef500153393f0',
}
export const productsMock = [
  {
    _id: '607057d9e448c200152a5047',
    cartQuantity: 1,
    product: productMock
  },
  {
    _id: '607057d9e448c200152a5048',
    cartQuantity: 1,
    product: productMock2
  }
];