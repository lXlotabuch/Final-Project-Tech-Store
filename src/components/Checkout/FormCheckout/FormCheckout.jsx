/* eslint-disable react/jsx-props-no-spreading */
import React, {
  useMemo, useRef, useState
} from 'react';
import PropTypes from 'prop-types';
import {
  Form, Input, Radio, Select
} from 'antd';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom/';
import {
  selectBranches,
  selectCities,
  selectCustomer,
  selectProducts,
  selectShippingCost
} from '../../../store/cart/reducer';
import {StyledRadio, StyledShippingTitle} from '../StyledCheckout';
import StyledButton from '../../common/Buttons/StyledButton';
import { getBranches, getShippingCost, placeOrder} from '../../../store/cart/middleware';
import { validName, validTelephone } from '../../../utils/constants';
import { layoutFormCheckout } from './utils';
    
export const FormCheckoutComponent = (props) => {
  const {
    cities,
    branches,
    customer,
    getBranches,
    shippingCost,
    placeOrder,
    products,
    getShippingCost
  } = props

  const { Option } = Select;
  const history = useHistory()
    
  const formLayout = layoutFormCheckout;
          
  const fields = useMemo(() => ([{
    name: 'email',
    value: customer.email || null
  },
  {
    name: 'firstName',
    value: customer.firstName || null
  },
  {
    name: 'lastName',
    value: customer.lastName || null
  },
  {
    name: 'phoneNumber',
    value: customer.telephone || '+380'
  },
  {
    name: 'country',
    value: 'Ukraine'
  },
  ]), [customer])
  
  const recipientCityRef = useRef();
  const countryRef = useRef();
  const branchSelect = useRef();
  const [valuePaymentInfo] = useState(
    'Payment at the time of receipt of the goods'
  );
  
  const [form] = Form.useForm();
    
  const handleCityChange = (cityRef) => {
    form.setFieldsValue({recipientBranch: null})
    getBranches(cityRef);
  }
  
  const onFinish = (values) => {
    placeOrder(products, values, customer, shippingCost, valuePaymentInfo)
    history.push('/order')
  };
      
  return (
    <Form
      {...formLayout}
      name="checkout-form"
      fields={fields}
      form={form}
      onFinish={onFinish}
    >
      <StyledShippingTitle>
        Customer Details:
      </StyledShippingTitle>
  
      <Form.Item
        label="Email"
        name="email"
        rules={[
          {
            required: true,
            message: 'Please input your active email.',
          },
          {
            type: 'email',
            message: 'Entered data is not an email.',
          },
        ]}
      >
        <Input
          placeholder="test@testmail.com"
        />
      </Form.Item>
        
      <Form.Item
        label="First name"
        name="firstName"
        rules={[
          {
            required: true,
            message: 'Please input your name.',
          },
          {
            type: 'string',
            min: 2,
            max: 25,
            message: 'FirstName must be between 2 and 25 letters',
          },
          {
            pattern: validName,
            message: 'First name cannot contain characters, space or numbers'
          },
        ]}
      >
        <Input placeholder="First name" />
      </Form.Item>
  
      <Form.Item
        label="Last name"
        name="lastName"
        rules={[
          {
            required: true,
            message: 'Please input your Last name.',
          },
          {
            type: 'string',
            min: 2,
            max: 25,
            message: 'LastName must be between 2 and 25 letters',
          },
          {
            pattern: validName,
            message: 'Last name cannot contain characters, space or numbers'
          },
        ]}
      >
        <Input placeholder="Last name" />
      </Form.Item>
  
      <Form.Item
        label="Phone number"
        name="phoneNumber"
        rules={[
          {
            required: true,
            message: 'Please input your phone number +380 XX XXX XXXX',
            min: 13,
            max: 13,
          },
          {
            pattern: validTelephone,
            message: 'Phone number must start with "+380", allowed characters is 0-9'
          }
        ]}
      >
        <Input placeholder="Mobile Number +380 XX XXX XXXX" />
      </Form.Item>
  
      <StyledShippingTitle>
        Payment Details:
      </StyledShippingTitle>
  
      <Radio.Group
        name="paymentInfo"
        value={valuePaymentInfo}
        style={{marginBottom: '20px'}}
      >
        <StyledRadio value="Payment at the time of receipt of the goods">
          Payment at the time of receipt of the goods
        </StyledRadio>
      </Radio.Group>
        
      <StyledShippingTitle>
        Shipping Details:
      </StyledShippingTitle>
  
      <Form.Item
        label="Country"
        name="country"
        rules={[{ required: true, message: 'Country is required' }]}
      >
        <Input disabled ref={countryRef} />
      </Form.Item>
  
      <Form.Item
        label="City"
        name="recipientCity"
        title="City choice"
        rules={[
          { required: true, message: 'Recipient city is required' },
            
        ]}
      >
        <Select
          placeholder="Select the city of recipient"
          onChange={handleCityChange}
          ref={recipientCityRef}
        >
          {cities.map((item) => (
            <Option value={item.Ref} key={item.Ref}>
              {item.CityName}
            </Option>
          ))}
        </Select>
      </Form.Item>
  
      <Form.Item
        label="â„– branch"
        name="recipientBranch"
        rules={[
          { required: true, message: 'Branch is required' },
        ]}
      >
        <Select
          placeholder="Select the branch of Nova Poshta of the recipient"
          onChange={() => getShippingCost(recipientCityRef)}
          ref={branchSelect}
        >
          {branches.map((item) => (
            <Option value={item.branchName} key={item.branchRef}>
              {item.branchName}
            </Option>
          ))}
        </Select>
      </Form.Item>
        
      <StyledButton shape="round" htmlType="submit">
        Place Order
      </StyledButton>
    </Form>
  )
}

const mapStateToProps = (state) => ({
  cities: selectCities(state),
  branches: selectBranches(state),
  customer: selectCustomer(state),
  shippingCost: selectShippingCost(state),
  products: selectProducts(state),
})

const FormCheckout = connect(
  mapStateToProps,
  {
    getBranches,
    getShippingCost,
    placeOrder
  }
)(FormCheckoutComponent)

FormCheckoutComponent.propTypes = {
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      CityName: PropTypes.string.isRequired,
      Ref: PropTypes.string.isRequired,
    })
  ).isRequired,
  branches: PropTypes.arrayOf(PropTypes.shape({
    branchName: PropTypes.string.isRequired,
    branchRef: PropTypes.string.isRequired,
  })).isRequired,
  shippingCost: PropTypes.number.isRequired,
  getBranches: PropTypes.func.isRequired,
  getShippingCost: PropTypes.func.isRequired,
  placeOrder: PropTypes.func.isRequired,
  customer: PropTypes.shape({
    telephone: PropTypes.string,
    lastName: PropTypes.string,
    firstName: PropTypes.string,
    email: PropTypes.string,
  }).isRequired,
  products: PropTypes.arrayOf(PropTypes.shape({
    imageUrls: PropTypes.arrayOf(PropTypes.string),
    name: PropTypes.string,
    currentPrice: PropTypes.number,
    _id: PropTypes.string,
  })).isRequired,
}

export default FormCheckout;
