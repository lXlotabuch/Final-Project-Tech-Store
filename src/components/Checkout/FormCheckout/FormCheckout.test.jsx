/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import {
  fireEvent, render, screen, waitFor
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { FormCheckoutComponent } from './FormCheckout';
import { productsMock } from '../../../mocks/products';
import { citiesMock } from '../../../mocks/cities';
import { branchesMock } from '../../../mocks/branches';
import { machMedia } from '../../../mocks/matchMedia.mock';

describe('FormCheckoutComponent', () => {
  beforeEach(() => {
    machMedia()
  })
  test('render', () => {
    const {asFragment} = render(<FormCheckoutComponent
      products={productsMock}
      shippingCost={121}
      customer={{
        customerNo: '82724063',
        email: 'o.titova.ua@gmail.com',
        enabled: true,
        firstName: 'test',
        isAdmin: true,
        lastName: 'test',
        telephone: '+380660788828',
        _id: '602ff192ad29b70015add9ab'
      }}
      branches={branchesMock}
      cities={citiesMock}
      getBranches={() => {}}
      placeOrder={() => {}}
      getShippingCost={() => {}}
    />)
    expect(asFragment()).toMatchSnapshot()
  });

  test('render with empty customer', () => {
    render(<FormCheckoutComponent
      products={productsMock}
      shippingCost={121}
      customer={{}}
      branches={branchesMock}
      cities={citiesMock}
      getBranches={() => {}}
      placeOrder={() => {}}
      getShippingCost={() => {}}
    />)
    expect(screen.getByLabelText(/email/i).value).toBe('')
    expect(screen.getByLabelText(/first name/i).value).toBe('')
    expect(screen.getByLabelText(/last name/i).value).toBe('')
    expect(screen.getByLabelText(/phone number/i).value).toBe('+380')
  });

  it('handle city change', async () => {
    const getBranches = jest.fn();
    render(<FormCheckoutComponent
      products={productsMock}
      shippingCost={121}
      customer={{
        customerNo: '82724063',
        email: 'o.titova.ua@gmail.com',
        enabled: true,
        firstName: 'test',
        isAdmin: true,
        lastName: 'test',
        telephone: '+380660788828',
        _id: '602ff192ad29b70015add9ab'
      }}
      branches={[]}
      cities={citiesMock}
      getBranches={getBranches}
      placeOrder={() => {}}
      getShippingCost={() => {}}
    />)

    fireEvent.mouseDown(screen.getByLabelText(/city/i))

    await waitFor(() => screen.getByTitle('Kyiv'))

    fireEvent.click(screen.getByTitle('Kyiv'))

    expect(getBranches).toBeCalledWith('8d5a980d-391c-11dd-90d9-001a92567626')
  });

  it('handle branch change', async () => {
    const getShippingCost = jest.fn();
    render(<FormCheckoutComponent
      products={productsMock}
      shippingCost={121}
      customer={{
        customerNo: '82724063',
        email: 'o.titova.ua@gmail.com',
        enabled: true,
        firstName: 'test',
        isAdmin: true,
        lastName: 'test',
        telephone: '+380660788828',
        _id: '602ff192ad29b70015add9ab'
      }}
      branches={branchesMock}
      cities={citiesMock}
      getBranches={() => {}}
      placeOrder={() => {}}
      getShippingCost={getShippingCost}
    />)

    fireEvent.mouseDown(screen.getByLabelText(/city/i))

    let cityEl;
    await waitFor(() => cityEl = screen.getByTitle('Kyiv'))

    fireEvent.click(cityEl)

    fireEvent.mouseDown(screen.getByLabelText(/№ branch/i))

    let branchEl;
    await waitFor(() => branchEl = screen.getByTitle('Branch name'))

    fireEvent.click(branchEl)

    expect(getShippingCost).toBeCalled();

    expect(getShippingCost.mock.calls[0][0]).toMatchObject({
      current: {
        props: {
          value: '8d5a980d-391c-11dd-90d9-001a92567626'
        }
      }
    })
  });

  it('handle put order', async () => {
    const placeOrder = jest.fn();
    const history = createMemoryHistory();

    render(
      <Router history={history}>
        <FormCheckoutComponent
          products={productsMock}
          shippingCost={121}
          customer={{
            customerNo: '82724063',
            email: 'o.titova.ua@gmail.com',
            enabled: true,
            firstName: 'test',
            isAdmin: true,
            lastName: 'test',
            telephone: '+380660788828',
            _id: '602ff192ad29b70015add9ab'
          }}
          branches={branchesMock}
          cities={citiesMock}
          getBranches={() => {}}
          placeOrder={placeOrder}
          getShippingCost={() => {}}
        />
      </Router>
    )
    fireEvent.mouseDown(screen.getByLabelText(/city/i))
    let cityEl;
    await waitFor(() => cityEl = screen.getByTitle('Kyiv'))
    fireEvent.click(cityEl)
    fireEvent.mouseDown(screen.getByLabelText(/№ branch/i))
    let branchEl;
    await waitFor(() => branchEl = screen.getByTitle('Branch name'))
    fireEvent.click(branchEl)
    fireEvent.click(screen.getByRole('button', {name: /place order/i}))
    await waitFor(() => expect(placeOrder).toBeCalled())
    expect(placeOrder.mock.calls[0]).toEqual([
      productsMock,
      {
        country: 'Ukraine',
        email: 'o.titova.ua@gmail.com',
        firstName: 'test',
        lastName: 'test',
        phoneNumber: '+380660788828',
        recipientBranch: 'Branch name',
        recipientCity: '8d5a980d-391c-11dd-90d9-001a92567626',
      },
      {
        _id: '602ff192ad29b70015add9ab',
        customerNo: '82724063',
        email: 'o.titova.ua@gmail.com',
        enabled: true,
        firstName: 'test',
        isAdmin: true,
        lastName: 'test',
        telephone: '+380660788828',
      },
      121,
      'Payment at the time of receipt of the goods',
    ])

    expect(history.location.pathname).toBe('/order')
  });
})
