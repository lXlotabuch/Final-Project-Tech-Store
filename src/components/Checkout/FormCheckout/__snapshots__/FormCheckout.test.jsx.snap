// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCheckoutComponent render 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    id="checkout-form"
  >
    <p
      class="sc-bdfBwQ jPpEGH"
    >
      Customer Details:
    </p>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_email"
          title="Email"
        >
          Email
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="checkout-form_email"
              placeholder="test@testmail.com"
              type="text"
              value="o.titova.ua@gmail.com"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_firstName"
          title="First name"
        >
          First name
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="checkout-form_firstName"
              placeholder="First name"
              type="text"
              value="test"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_lastName"
          title="Last name"
        >
          Last name
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="checkout-form_lastName"
              placeholder="Last name"
              type="text"
              value="test"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_phoneNumber"
          title="Phone number"
        >
          Phone number
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="checkout-form_phoneNumber"
              placeholder="Mobile Number +380 XX XXX XXXX"
              type="text"
              value="+380660788828"
            />
          </div>
        </div>
      </div>
    </div>
    <p
      class="sc-bdfBwQ jPpEGH"
    >
      Payment Details:
    </p>
    <div
      class="ant-radio-group ant-radio-group-outline"
      style="margin-bottom: 20px;"
    >
      <label
        class="ant-radio-wrapper ant-radio-wrapper-checked sc-dlfnbm NRStK"
      >
        <span
          class="ant-radio ant-radio-checked"
        >
          <input
            checked=""
            class="ant-radio-input"
            name="paymentInfo"
            type="radio"
            value="Payment at the time of receipt of the goods"
          />
          <span
            class="ant-radio-inner"
          />
        </span>
        <span>
          Payment at the time of receipt of the goods
        </span>
      </label>
    </div>
    <p
      class="sc-bdfBwQ jPpEGH"
    >
      Shipping Details:
    </p>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_country"
          title="Country"
        >
          Country
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input ant-input-disabled"
              disabled=""
              id="checkout-form_country"
              type="text"
              value="Ukraine"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
      title="City choice"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_recipientCity"
          title="City"
        >
          City
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="checkout-form_recipientCity_list_0"
                    aria-autocomplete="list"
                    aria-controls="checkout-form_recipientCity_list"
                    aria-haspopup="listbox"
                    aria-owns="checkout-form_recipientCity_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="checkout-form_recipientCity"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  Select the city of recipient
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 ant-col-md-8 ant-col-lg-7"
      >
        <label
          class="ant-form-item-required"
          for="checkout-form_recipientBranch"
          title="№ branch"
        >
          № branch
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16 ant-col-md-16 ant-col-lg-12"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="checkout-form_recipientBranch_list_0"
                    aria-autocomplete="list"
                    aria-controls="checkout-form_recipientBranch_list"
                    aria-haspopup="listbox"
                    aria-owns="checkout-form_recipientBranch_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="checkout-form_recipientBranch"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                >
                  Select the branch of Nova Poshta of the recipient
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="ant-btn ant-btn-round sc-hKgILt eIYqNo"
      type="submit"
    >
      <span>
        Place Order
      </span>
    </button>
  </form>
</DocumentFragment>
`;
